woodwork/type_sys/inference_functions.py

```diff
@@ -93,8 +94,18 @@ def boolean_nullable_func(series):
                 {False},
             ]:
                 return True
+            series_lower = set(str(s).lower() for s in set(series))
+            if series_lower in [
+                set(boolean_list)
+                for boolean_list in config.get_option("boolean_inference_strings")
+            ]:
```

Comment:
So currently if only `True` OR `False` (instead of both) is in a series, then it can be identified as `Boolean` (lines 91 to 94 above). This logic here says that both the positive and negative class (both "y" and "n", or "0" and "1") have to be present in the series for it to be classified as a Boolean match. Is this something we're ok with?

Reply:
Yeah I think it would need to contain exactly 1 or both of the items in the set. So a full column of "true" should still be boolean. Particularly because we're still selecting from the head.

Reply:
@bchen1116 I think the problem here is that we're sampling from the head and it's entirely possible that the first 100k rows are all true while the last 100k are all false or something similar. If we were doing a fully random sample, Id say you're probably right, but while we're looking at the HEAD only it might be a bit risky. Ultimately, probably not a dealbreaker since Parthiv will be working on that soon anyway, but something to think about. And currently typelib works in that if only 1 appeared it would still parse as boolean.

Reply:
@ParthivNaresh Regarding your example of only `True` or only `False` Python objects, if that is the case, the user has most likely done their own casting (pre-processing). Therefore, it makes sense for us to assume that it's a Boolean Logical Type.

With the string representation of only `yes` or only `no`, it seems presumptuous to assume that it's a Boolean Logical Type. What if there was an another category of `maybe`? 

I originally created the issue for this implementation (https://github.com/alteryx/woodwork/issues/153) because I had seen a column full of `yes`, `no` in a dataset. I have never seen a column with only `yes` or only `no`. 
Given Parthiv's upcoming work, maybe we punt on this? Additionally, if we get an actual dataset where this is problematic, we can revisit?

Reply:
Yeah I think this is fine. Sampling should be in probably within 2 sprints, so I think I'm okay with punting on this edge case for now. Though I'd say we probably should be consistent across all string representations. In which case, with a randomly sampled dataset, this logic should probably be fine.